### DISCUSSION TITLE
Get logarhithmic scale on x and y

### DISCUSSION THREAD

**Discussion by gernophil** [whitews/FlowKit#172] [Q&A]
Hey everyone,
I'm pretty new to `FlowKit`. Tried `FlowCytometryTools` before, but this seems a bit outdated. So far I really like it. There is just on thing that I did not find in the tutorials and also not online anywhere. How can I get logarithmic scales on the x and y axis? The same question was asked [here](https://discourse.bokeh.org/t/how-to-make-scatter-plots-with-log-scale-on-both-axes/6745/6) but without any answers.
Thanks everyone

**Comment by whitews** [comment]
Hi,

Correct me if mistaken, but I am assuming you mean log scale in the untransformed space for transformed events. Or, more simply, why don't the x & y axis scales look like FlowJo?

The reason for this has to do with FlowJo being a GUI where the user can point and click versus a toolkit with a Python API. Gate boundaries, like polygon vertices, must be given in the transformed space. Because of this, the decision was made to display the scales exactly as the transform dictates and not obfuscate the positions where the boundaries should be defined. 

For a GUI like FlowJo, the user doesn't need to know how this works, they click their polygon vertices by looking at the plot. For FlowKit, imagine trying to define a polygon gate from looking at a plot of untransformed log scaled axes. You would have to mentally transform every coordinate to define your gate.

A similar logic applies to getting transformed event data via FlowKit. It would be somewhat confusing to retrieve those NumPy arrays and see the actual transformed values that do not correspond to the plots that FlowKit creates.

Hope this clarifies things and let me know if I'm misreading your question,
Scott

**Reply by gernophil** [reply]
Yes, you are right about my intentions. And I totally understand your point, but in the end I guess what everyone needs is a nice figure (after all the analysis of course). And wouldn't it be great, if this could be done just by FlowKit and then with logarithmic axes?

**Reply by whitews** [reply]
I've had this conversation with others but haven't received a compelling reason for why this is useful beyond "It's the way FlowJo does it". I'd consider a PR if you want to have a go, but currently there are many other updates & features I want to tackle when I have the time to work on FlowKit.

**Reply by gernophil** [reply]
Well, it's not like FlowJo "invented" logarithmic scaling. Logarithmic scaling is not even restricted to flow cytometry of biology. It's widely used all across science, if you wanna plot "exponential" data, but I guess you are aware of this. There's not much of a difference between transforming data or transforming the scale. In the end they're basically the same (and I know, if we would start arguing I would loose, because I lack deep enough knowledge of mathematics). Just on a logarithmic scale you retain the actual raw data and with transforming data you retain the cartesian ccordinates (not sure, if the term is correct here, but I guess you know what I mean). I am pretty new to flowcyt, but I prefer log scale since it gives me a better feeling for the real numbers. And I know the feeling for numbers is bad in general for humans. I guess you would find enough arguments that log scaling would even be more natural (and the other way around), but in the end it's more of a question of taste and more of a philosophic question then a question of "it's what FlowJo" does. Why not just let the user decide what he likes more :). And of course, implementing new stuff is always a question of time and workload. Maybe I'll give it a try. Thanks for this great package and your effort so far :).
